src/main/java/uk/gov/hmcts/ccd/data/casedetails/DefaultCaseDetailsRepository.java

```diff
@@ -75,9 +77,16 @@ public CaseDetails set(final CaseDetails caseDetails) {
             throw new CaseConcurrencyException("The case data has been altered outside of this transaction.");
         } catch (PersistenceException e) {
             LOG.warn("Failed to store case details", e);
-            if (e.getCause() instanceof ConstraintViolationException) {
-                throw new CaseConcurrencyException(e.getMessage());
+
+            if (e.getCause() instanceof ConstraintViolationException
+                && ((ConstraintViolationException) e.getCause()).getConstraintName()
+                .equals(UNIQUE_REFERENCE_KEY_CONSTRAINT)) {
```

Comment:
for readability can you move this into a private method `isDuplicateReference(e)` or similar please